### PR TITLE
:sparkles: Add LemLib Odometry Code Generator v0.4.x

### PR DESCRIPTION
**Pull request by sufferiing**
This is moved from #10 

Add a format that exports just the coordinate points as LemLib odometry code, ignoring heading, speed, and interpolated points. The points are converted to be relative to the starting point (as is default in LemLib). The format export generates a series of moveTo calls that can be pasted directly into code.

### CODE REVIEW

File: src/format/Format.tsx [hunk 1]
```diff
@@ -42,7 +43,8 @@ export interface Format extends ExecutionEventListenersContainer<CancellableComm
 export function getAllFormats(): Format[] {
   return [
     new LemLibFormatV0_4(), //
-    new PathDotJerryioFormatV0_1()
+    new PathDotJerryioFormatV0_1(),
```
Comment (Jerrylum) [inline src/format/Format.tsx:46]: Change the order to:
```
new LemLibFormatV0_4(), //
new LemLibOdomGeneratorFormatV0_1(),
new PathDotJerryioFormatV0_1()
```

**Review by Jerrylum — CHANGES_REQUESTED** [review]

**Comment by BattleCh1cken** [comment]
I don't really know enough about the codebase to make relevant comments there, but I pulled down the PR and messed around with it for a bit. I think this idea is really cool, but it'd be nice to have some kind of option to like save to clipboard or something. The workflow I have right now is to save the file, open it, and then copy and paste the code from there anyway.

Basically just any way to get the generated code into my codebase faster would be great.

**Comment by Jerrylum** [comment]
Please fix the merge conflicts

**Comment by SizzinSeal** [comment]
So, couple of things. Odom generator format should be V0.4, not V0.1. Reason for this is because odom generator output will only work for V0.4, not other (upcoming) versions.

The output does not provide code that compiles. It is missing the required `timeout` parameter. For example
```
chassis.moveTo(x, y);
// should be
chassis.moveTo(x, y, timeout);
```
Timeout can just default to 5000

**Review by BattleCh1cken — APPROVED** [review]

File: src/format/LemLibOdomGeneratorFormatV0_4.tsx [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+            getValue={() => this.movementTimeout.toString() }
+            setValue={(value: string) => {
+              const parsedValue = parseInt(value)!;
+              app.history.execute(
+                `Change default movement timeout`,
+                new UpdateProperties(this as any, { movementTimeout: parsedValue })
+              );
+            }}
+            isValidIntermediate={() => true}
+            isValidValue={(candidate: string) => parseInt(candidate) !== undefined}
```
Comment (Jerrylum) [inline src/format/LemLibOdomGeneratorFormatV0_4.tsx:117]: 1. Attribute `numeric` should be added to the `ObserverInput`.

2. Error occurs when calling `parseInt("")`, which is equal to `NaN`. Please check out `ControlAccordion` line 18 to learn more about how `ObserverInput` should handle number input. You better use `Tokens.Int.parse` to parse the value.

File: src/format/LemLibOdomGeneratorFormatV0_4.tsx [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+      }
+
+      // ALGO: Offsets to convert the absolute coordinates to the relative coordinates LemLib uses
+      const offsets = new Vector(start.x, start.y);
+      for (const point of points) {
+        // ALGO: Only coordinate points are supported in LemLibOdom format
+        const relative = euclideanRotation(heading, point.subtract(offsets));
+        rtn += `${gc.chassisName}.moveTo(${uc.fromAtoB(relative.x).toUser()}, ${uc.fromAtoB(relative.y).toUser()}, ${gc.movementTimeout});\n`;
+      }
+    }
```
Comment (Jerrylum) [inline src/format/LemLibOdomGeneratorFormatV0_4.tsx:238]: How are you going to recover the path file if `#PATH.JERRYIO-DATA` line is not included? Or do you think it is not needed?

File: src/format/LemLibOdomGeneratorFormatV0_4.tsx [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+    makeAutoObservable(this);
+  }
+
+  getConfigPanel() {
+    return <></>;
+  }
+}
+
+// observable class
+export class LemLibOdomGeneratorFormatV0_1 implements Format {
```
Comment (Jerrylum) [inline src/format/LemLibOdomGeneratorFormatV0_4.tsx:168]: Rename it to V0_4

File: src/core/InputOutput.ts [hunk 5]
```diff
@@ -313,6 +313,12 @@ export async function onDownloadAs(fallback: boolean = false): Promise<boolean>
   return true;
 }
 
+export async function onCopyToClipboard(): Promise<boolean> {
+  const code = exportPathFile()!;
```
Comment (Jerrylum) [inline src/core/InputOutput.ts:317]: You can return false if `exportPathFile` returned undefined. Don't use `!` here.

**Review by Jerrylum — CHANGES_REQUESTED** [review]

File: src/format/LemLibOdomGeneratorFormatV0_4.tsx (hunk 3 above)
Comment (sufferiing) [reply src/format/LemLibOdomGeneratorFormatV0_4.tsx:240]: Need to change onClipboardCopy to be specific to this format so we can add the `#PATH.JERRYIO-DATA` section without including it in the clipboard code.

File: src/format/LemLibOdomGeneratorFormatV0_4.tsx [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+            }}
+            isValidIntermediate={() => true}
+            isValidValue={(candidate: string) => candidate !== ""}
+            sx={{ marginTop: "16px" }}
+          />
+          <ObserverInput
+            label="Movement Timeout"
+            getValue={() => this.movementTimeout.toString() }
+            setValue={(value: string) => {
+              const parsedValue = parseInt(value);
```
Comment (Jerrylum) [inline src/format/LemLibOdomGeneratorFormatV0_4.tsx:111]: should also use `Int.parse(new CodePointBuffer(candidate))`

File: src/app/MenuAccordion.tsx [hunk 7]
```diff
@@ -463,9 +463,16 @@ const FileMenuItems = () => {
       key={makeId(10)}
       showLeftIcon={false}
       label="Download As"
-      hotkey="Shift+Mod+D"
+      hotkey="Mod+Shift+D"
       onClick={() => onDownloadAs()}
     />,
+    <CustomMenuItem
```
Comment (Jerrylum) [inline src/app/MenuAccordion.tsx:469]: Ref 1

File: src/core/InputOutput.ts [hunk 8]
```diff
@@ -313,6 +313,15 @@ export async function onDownloadAs(fallback: boolean = false): Promise<boolean>
   return true;
 }
 
+export async function onCopyToClipboard(): Promise<boolean> {
```
Comment (Jerrylum) [inline src/core/InputOutput.ts:316]: Ref 2

**Review by Jerrylum — CHANGES_REQUESTED** [review]
As discussed in the discord, you can remove function ref 1 and ref 2, I will merge this PR to the main and take over the hotkey feature.

File: src/format/LemLibOdomGeneratorFormatV0_4.tsx (hunk 3 above)
Comment (Jerrylum) [reply src/format/LemLibOdomGeneratorFormatV0_4.tsx:240]: The `#PATH.JERRYIO-DATA` section should be added as discussed on Discord, see the next change request.

**Review by Jerrylum — CHANGES_REQUESTED** [review]

File: .prettierignore [hunk 9]
```diff
@@ -1,16 +1,15 @@
...
-/package.json
-/src/app/MarkdownSupport.tsx
-/src/app/TestMarkdown.mdx
-/src/app/Welcome.mdx
-/src/react-app-env.d.ts
-/src/setupTests.ts
-/src/token/Tokens.test.ts
-/tsconfig.json
\ No newline at end of file
+.github/ISSUE_TEMPLATE/bug_report.md
```
[the remark this answers is not in the document]
Comment (sufferiing) [reply .prettierignore:1]: It was a mistake, I thought I had reverted it.

**Review by Jerrylum — APPROVED** [review]
Gonna take over

**Pull request merged**